EU Trusted Lists Section

cmarco0 opened the pull request:
this PR aims to close #258 

this answer 2 questions:

-how a WI can trust the WP
-how a WP can trust the WI

cmarco0 committed:
Update Wallet Attestation process 

how a WP can trust the WI and the other way round

peppelinux commented on docs/en/wallet-attestation.rst:
```diff
@@ -59,6 +59,12 @@ Wallet Instance Initialization and Registration
 
+.. note::
+
+  The WP MUST verify the WI by using the app store vendors API, for Android is Play Integrity API and for iOS is DeviceCheck, these services are defined in this specification as **Device Integrity Service (DIS)**.
```
a requirement must not be defined in a note but within the list of the requirements

this can be only applied to local integrated WSCD, therefore this is something to be specialized in that context and not widely

cmarco0 replied:
I can append it to the fourth requirement already defined.

peppelinux replied:
please do.

peppelinux commented on docs/en/wallet-attestation.rst:
```diff
@@ -68,6 +74,10 @@ Wallet Instance Initialization and Registration
 
+.. note::
+
+  The Trust Framework works in a way that the Wallet Providers though their Trust Chain Root Authorities are anchored in Trust List managed by appointed Supervisory Body or  by a delegated authority. This ensures the integrity and authenticity of wallet solutions are rigorously maintained. These layers of security and oversight create a trusted environment, allowing users to rely on the legitimac ...
```
double spaces, missing defined terminology

generally the trust frameworks defines, doesn't operationally works.

Please avoid qualifying adjectives like "robust", keep a linear language carring technical requirements and evidences

cmarco0 committed:
Editorial change in the note, trust framework defines and not work

peppelinux commented on docs/en/wallet-attestation.rst:
```diff
@@ -68,6 +74,10 @@ Wallet Instance Initialization and Registration
 
+.. note::
+
+  The Trust Framework defines that the Wallet Providers though their Trust Chain Root Authorities are anchored in Trust List managed by appointed Supervisory Body or  by a delegated authority. This ensures the integrity and authenticity of wallet solutions are rigorously maintained. These layers of security and oversight create a trusted environment, allowing users to rely on the legitimacy and s ...
```
part of this must be moved in the section about the trust infrastructure, where we must bring more details about the integrations with superior trust lists and the implementaiton of the national ones

only the part strictly related to the wallert solution can be kept here

cmarco0 committed:
update trust list infrastructure section

details about superior trust lists and the national ones

cmarco0 committed:
update the part strictly related to the wallert solution

peppelinux commented on docs/en/trust.rst:
```diff
@@ -578,6 +578,8 @@ Trust Chain
 
 The Trust Chain is a sequence of verified statements that validates a participant's compliance with the Federation. It has an expiration date time, beyond which it MUST be renewed to obtain the fresh and updated metadata. The expiration date of the Trust Chain is determined by the earliest expiration timestamp among all the expiration timestamp contained in the statements. No Entity can force the ...
 
+The Trust Framework establishes that Wallet Providers, through their Trust Chain Root Authorities, are anchored in a Trust List managed by an appointed Supervisory Body or a delegated authority. This multi-layered security and oversight system creates a reliable and secure environment, ensuring users can trust the legitimacy and safety of their wallet instances.
```
```suggestion
The Wallet Providers MUST be published in a Trust List managed by the designed Federation authority. 
```

The Trust List is out of the Federation Trust Chain. 

Any reference to the Trust List MUST be handled in a specialized section, and therefore out from this section defined for the openid federation trust chain.

peppelinux commented on docs/en/trust.rst:
```diff
@@ -602,6 +604,13 @@ Below is a non-normative example of a Trust Chain in its original format (JSON A
 
     The entire Trust Chain is verifiable by only possessing the Trust Anchor's public keys.
 
+**Trust List implementation**
```
see my previous comment

peppelinux commented on docs/en/wallet-attestation.rst:
```diff
@@ -59,6 +59,12 @@ Wallet Instance Initialization and Registration
+
+  The WP MUST verify the WI by using the app store vendors API, for Android is Play Integrity API and for iOS is DeviceCheck, these services are defined in this specification as **Device Integrity Service (DIS)**.
+  The EUDIW Application MUST also implement the integrity services from the vendor's SDK, this service has already beed defined as **Device Integrity Service (DIS)** in this specification. The DIS helps by detecting potentially risky and fraudulent interactions, such as from tampered app versions and untrustworthy environments.
+The verification process to establish the trustworthiness of a WI for the WP begins with the initial app launch during which a GUID is generated, serving as the WI's identifier. During the *Initialization and Registration* process, the WI transmits this GUID to the WP, which in turn generates a key pair and signs the GUID with the private key. This signed GUID is then retained by the WI. Subseque ...
```
where this GUID is defined?

cmarco0 replied:
i'll change it to UUDI.

It is not defined in the specification.

A UUID, generated by the WSCD, serves as a unique identifier for the WI and helps prevent attacks or data manipulation. In this context, the WP can use the UUID to establish trust with a WI.

grausof replied:
I think you are referring to: `Cryptographic Hardware Key Tag` 
https://github.com/italia/eudi-wallet-it-docs/blob/18a09ac4ad54bc7110259f249526acc6286c648b/docs/en/defined-terms.rst#defined-terms

cmarco0 replied:
No, the Cryptographic Hardware Key Tag is linked to the generated Cryptographic Hardware Keys and is used to access the private key stored in the hardware.

What I'm referring to is a unique identifier generated by the OS when the WI is launched for the first time, and it exclusively identifies the WI.

grausof replied:
In our implementation the hardware key tag is also the unique identifier of the Wallet Instance.

> As mentioned in the *Wallet Instance initialization and registration* section above, a Wallet Instance is bound to a Wallet Hardware Key and it's uniquely identified by it.

cmarco0 replied:
So in this case, i can replace UUID with cryptographic hardware key tag.
What do you think ?

cmarco0 committed:
editorial change

the sentence about the WI verification from WP is a requirement and has beed moved in that section.

cmarco0 committed:
editorial change

Co-authored-by: Giuseppe De Marco <giuseppe.demarco@teamdigitale.governo.it>

cmarco0 committed:
Edit from GUID to UUID

cmarco0 committed:
editorial update

replace UUID with Cryptographic Hardware Key Tag

cmarco0 committed:
editorial update

cmarco0 committed:
editorial update

peppelinux commented on docs/en/wallet-attestation.rst:
```diff
@@ -75,7 +76,7 @@ The verification process to establish the trustworthiness of a WI for the WP beg
 .. note::
 
-As explained in the chapter `Trust Infrastructure <https://github.com/cmarco0/eudi-wallet-it-docs-fork/blob/cmarco0-DynamicComponentViewProcess/docs/en/trust.rst#the-infrastructure-of-trust>`_  section `Trust Chain <https://github.com/cmarco0/eudi-wallet-it-docs-fork/blob/cmarco0-DynamicComponentViewProcess/docs/en/trust.rst#trust-chainit>`_ it ensures the integrity and authenticity of wallet sol ...
+    As explained in the chapter `Trust Infrastructure <https://github.com/cmarco0/eudi-wallet-it-docs-fork/blob/cmarco0-DynamicComponentViewProcess/docs/en/trust.rst#the-infrastructure-of-trust>`_  section `Trust Chain <https://github.com/cmarco0/eudi-wallet-it-docs-fork/blob/cmarco0-DynamicComponentViewProcess/docs/en/trust.rst#trust-chainit>`_ it ensures the integrity and authenticity of wallet ...
```
please use relative hrefs and not your personal github repo

cmarco0 committed:
editorial update href

cmarco0 committed:
editorial change; missing indent

cmarco0 committed:
created Trust List section

peppelinux commented on docs/en/trust.rst:
```diff
@@ -604,11 +602,16 @@ Below is a non-normative example of a Trust Chain in its original format (JSON A
+
 To ensure coherent and efficient management of trust lists across Europe, a structured approach has been proposed. This involves creating and governing a Superior Trust List at the European level and National Trust Lists at the member state level. The following sections provide the implementation details for each type of trust list.
 
 The **Superior Trust List** should be managed by a central entity at the European level, such as the European Commission. It will include direct references to each National Registry and each centrally managed Thematic Registry, unique for all member states. The governance is centralized under a single EU authority, authorized to add, remove, or update entries in the registry.
```
this sounds more like a proposal, while the scope of this technical specification is not to share proposal but to offer clear implementation configurations and examples. Not sure about this editorial cut

here I expect to get how the trust list must be implemented, the format used and the non normative examples about requests and responses

peppelinux commented on docs/en/trust.rst:
```diff
@@ -604,11 +602,16 @@ Below is a non-normative example of a Trust Chain in its original format (JSON A
 The **Superior Trust List** should be managed by a central entity at the European level, such as the European Commission. It will include direct references to each National Registry and each centrally managed Thematic Registry, unique for all member states. The governance is centralized under a single EU authority, authorized to add, remove, or update entries in the registry.
 
-The **National Trust List** should be managed by a national coordinating entity, ideally the National Supervisory Body or an entity delegated by it. This entity will receive requests from accredited and authoritative entities for the respective themes they manage. The Trust List will include direct references to each National List (Thematic, Wallet, TSP, and Devices Registries) and to the Superio ...
+The **National Trust List** should be managed by a national coordinating entity, ideally the National Supervisory Body or an entity delegated by it. This entity will receive requests from accredited and authoritative entities for the respective themes they manage. The Trust List will include direct references to each National List (thematic, Wallet, TSP, Devices Registries etc...) and to the Supe ...
```
should -> must

national coordinating entity is too much generic and therefore not actionable ... We need to use a clear terminology using an established role within the ecosystem. Please find it in the european regulations.

I don't like using `etc...` we need to enumerate all the entities required to be published within the trusted list. When this information is not clear and neither in our possession, we may think to explain one or more open points in the form of a note.

cmarco0 committed:
added non-normative example

cmarco0 committed:
Merge branch 'versione-corrente' into cmarco0-DynamicComponentViewProcess

cmarco0 committed:
editorial update

replace NAB with Supervisory Body

fmarino-ipzs reviewed (commented):
I have a couple of comments on this:
1. I would remove any real reference in terms of URL or Organizations etc. from the text. 
2. the trust list section is something that is not currently in the scope of this technical specification at this moment. I would postpone this to a future release.

peppelinux commented:
This section requires to be moved to a file dedicated to the european trusted lists

peppelinux commented:
@SaraConsoliACN this PR doesn't answer to the point raised here: https://github.com/italia/eudi-wallet-it-docs/issues/258#issuecomment-2091440030

SaraConsoliACN commented on docs/en/trust.rst:
```diff
@@ -628,6 +628,152 @@ Below is a non-normative example of a Trust Chain in its original format (JSON A
+Trust List
+^^^^^^^^^^^^^^^
+
+The Wallet Providers MUST be published in a Trust List managed by the designed Federation authority.
```
```suggestion
The Trust Chain including Wallet Provider MUST be anchored in a Trust List/Registry managed by the appointed Supervisory Body, where its Public Keys are available for validation purposes. It’s responsibility of the Supervisory Body to ensure that any Trust Chain anchored, contains exclusively certified Wallet Providers.
```

SaraConsoliACN replied:
@peppelinux Does this review allow the closure of issue #258?

peppelinux commented:
The trusted lists are available on the eIDAS Dashboard

https://eidas.ec.europa.eu/efda/home

peppelinux commented:
Closed by https://github.com/italia/eid-wallet-it-docs/pull/988
